Add/fix serializers for pandas types

jtyberg opened the pull request:
Add pandas Series serializer.

Switch to using `to_json` method for serializing pandas DataFrames.  Converts
dates and timestamps to integers.

Fixes #138

Commit:
Add serializer for pandas Series.

Use `to_json` method for serializing pandas DataFrames.  Converts
dates and timestamps to integers.

(c) Copyright IBM Corp. 2015

lbustelo commented:
@jtyberg Could you add a simple example of a function returning Series and rendering something in notebooks/examples/urth-core-function.ipynb?

lbustelo commented:
@jtyberg I also need you to rebase your branch. We have a temp fix for a bower issue that will cause problem with building if you don't update.

jtyberg commented:
During a conversation with @lbustelo, I realize that I completely missed the fact that the Serializers' `serialize` function actually returns a python object, NOT a JSON string, so don't merge this.

jtyberg commented:
I updated the Series/DataFrame serialization code to return an object instead of JSON.

Also added a couple of serialization examples to urth-core-function notebook.  Tested the examples on 3.2.x and 4.x.

wellecks commented:
Ran the serialization-related function examples successfully, looks good to me for handling Pandas types; agreed that we should re-think our Python serialization process for the general case

Commit:
Change default pandas datetime serialization to ISO format.

(c) Copyright IBM Corp. 2015

jtyberg commented:
Worked with @deedubbu on an existing dashboard notebook with the pandas serialization changes.  Passing Series and DataFrames between python and widgets is working fine.

Added a commit to change the default pandas serialization of timestamps to ISO formatted strings instead of epoch (milliseconds) so the widget author has to do less to see a human-readable date.

lbustelo commented on kernel-python/urth/util/serializers.py:
```diff
@@ -26,7 +28,28 @@
+        # Default to index orientation
+        # {index -> {column -> value}}
+        date_format = kwargs.get('date_format', 'iso')
+        return json.loads(obj.to_json(orient='index', date_format=date_format))
```
wouldn't you want to limit a series the same way we limit a dataframe?

jtyberg replied:
You might, except I chose to serialize the Series objects using the default `index` orientation, which basically gives you the equivalent of a `dict` in Python, or an `Object` in JavaScript.  So limiting in this sense is like saying, give me the first 100 attributes of the object.

I suppose if we wanted to change the default serialization to treat the data more like a long array, then the limit makes more sense.  

Either way, I can add the code to honor a limit.

lbustelo replied:
Nah... I think your comments makes sense... but might still be dangerous...

lbustelo commented:
Looks good and all my test passed. Thanks!

lbustelo commented:
I'm cherry picking these 2 commits into 0.2.x branch
